lib/Common.pm

```diff
@@ -2,32 +2,50 @@ package Common;
 
 use strict;
 use warnings;
+use warnings FATAL => 'all';
 
 use Exporter qw(import);
 
 # The below methods are used by build_java.pl and tests/all.pl; if the
 # location of Common.pm ever changes, update the includes in those files
 # to point to the new location.
-our @EXPORT_OK = qw(get_jar_files);
+our @EXPORT_OK = qw(get_jar_files detect_jar_file);
+
+sub detect_jar_file {
+    # Detect the correct path to a jar file.
+    # Die if the jar is missing.
+
+    my @locations = qw(/usr/share/java /usr/lib/java);
+    for my $location (@locations) {
+        for my $candidate (@_) {
+            my $path="$location/$candidate";
+            if (-e $path) {
+                return $path;
+            }
+        }
+    }
+
+    die "Can't find any jars for $_[1]";
+}
 
 sub get_jar_files {
```

Comment:
In the future it would be nice to have this accept an array of additional jars to find / search for. But it'd likely have to be an array of arrays which I'm not wanting to introduce that complexity right now. Instead I exported `detect_jar_file` so others callers (`tests/all.pl`) can use that.